diff --git a/pybind11_protobuf/proto_cast_util.cc b/pybind11_protobuf/proto_cast_util.cc
index 5ca857b..9cb889d 100644
--- a/pybind11_protobuf/proto_cast_util.cc
+++ b/pybind11_protobuf/proto_cast_util.cc
@@ -355,10 +355,10 @@ class PythonDescriptorPoolWrapper {
     // the Python DescriptorPool.
 
     // Find a file by file name.
-    bool FindFileByName(const std::string& filename,
+    bool FindFileByName(absl::string_view filename,
                         FileDescriptorProto* output) override {
       try {
-        auto file = pool_.attr("FindFileByName")(filename);
+        auto file = pool_.attr("FindFileByName")(std::string(filename));
         return CopyToFileDescriptorProto(file, output);
       } catch (py::error_already_set& e) {
         std::cerr << "FindFileByName " << filename << " raised an error";
@@ -371,10 +371,10 @@ class PythonDescriptorPoolWrapper {
     }
 
     // Find the file that declares the given fully-qualified symbol name.
-    bool FindFileContainingSymbol(const std::string& symbol_name,
+    bool FindFileContainingSymbol(absl::string_view symbol_name,
                                   FileDescriptorProto* output) override {
       try {
-        auto file = pool_.attr("FindFileContainingSymbol")(symbol_name);
+        auto file = pool_.attr("FindFileContainingSymbol")(std::string(symbol_name));
         return CopyToFileDescriptorProto(file, output);
       } catch (py::error_already_set& e) {
         std::cerr << "FindFileContainingSymbol " << symbol_name
@@ -389,11 +389,11 @@ class PythonDescriptorPoolWrapper {
 
     // Find the file which defines an extension extending the given message type
     // with the given field number.
-    bool FindFileContainingExtension(const std::string& containing_type,
+    bool FindFileContainingExtension(absl::string_view containing_type,
                                      int field_number,
                                      FileDescriptorProto* output) override {
       try {
-        auto descriptor = pool_.attr("FindMessageTypeByName")(containing_type);
+        auto descriptor = pool_.attr("FindMessageTypeByName")(std::string(containing_type));
         auto file =
             pool_.attr("FindExtensionByNumber")(descriptor, field_number)
                 .attr("file");
