diff --git a/third_party/xla/xla/service/custom_call_status_internal.h b/third_party/xla/xla/service/custom_call_status_internal.h
index e2ccc511871..f8f8d469e50 100644
--- a/third_party/xla/xla/service/custom_call_status_internal.h
+++ b/third_party/xla/xla/service/custom_call_status_internal.h
@@ -17,6 +17,7 @@ limitations under the License.
 #define XLA_SERVICE_CUSTOM_CALL_STATUS_INTERNAL_H_
 
 #include <optional>
+#include <string>
 
 #include "absl/strings/string_view.h"
 #include "xla/service/custom_call_status.h"
