diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index fb87fda9f82..c4f3d15db98 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -1336,7 +1336,11 @@ tf_cc_shared_library(
         "//tensorflow:macos": [
             "-Wl,-exported_symbols_list,$(location //tensorflow:tf_exported_symbols.lds)",
         ],
-        "//tensorflow:windows": [],
+        "//tensorflow:windows": [
+            "/WHOLEARCHIVE:tensorflow_framework.lib",
+            "/NODEFAULTLIB:libcmt.lib",
+            "/DEFAULTLIB:msvcrt.lib",
+        ],
         "//conditions:default": [
             "-z defs",
             "-Wl,--version-script,$(location //tensorflow:tf_version_script.lds)",
diff --git a/tensorflow/lite/acceleration/configuration/BUILD b/tensorflow/lite/acceleration/configuration/BUILD
index a8e13de931c..fa46c39dce0 100644
--- a/tensorflow/lite/acceleration/configuration/BUILD
+++ b/tensorflow/lite/acceleration/configuration/BUILD
@@ -46,7 +46,7 @@ genrule(
     # classes would have the same names.
     cmd = """
         $(location {}) --proto -o $(@D) $(location :configuration.proto)
-        perl -p -i -e 's/tflite.proto/tflite/' $@
+        perl -p -i.bak -e 's/tflite.proto/tflite/' $@
     """.format(flatc_path),
     compatible_with = get_compatible_with_portable(),
     tools = [flatc_path],