diff --git a/tensorflow/python/BUILD b/tensorflow/python/BUILD
index c3f9a377e01..98e594c6c79 100644
--- a/tensorflow/python/BUILD
+++ b/tensorflow/python/BUILD
@@ -1225,6 +1225,7 @@ tf_python_pybind_extension(
         "_pywrap_tfe.pyi",
     ],
     static_deps = tf_python_pybind_static_deps(),
+    win_def_file = ":pywrap_tensorflow_filtered_def_file",
     deps = [
         "//tensorflow/c:pywrap_required_hdrs",
         "//tensorflow/c/eager:tfe_tensorhandle_internal_hdrs_only",
@@ -1249,6 +1250,8 @@ tf_python_pybind_extension(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/types:optional",
+        "@com_google_absl//absl/status",
+        "//xla/hlo/ir:hlo",
         "@pybind11",
         # copybara:uncomment "@pybind11_protobuf//pybind11_protobuf:native_proto_caster",
     ] + if_static(
