diff --git a/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl b/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
--- a/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
+++ b/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
@@ -100,14 +100,11 @@
         "-emit-llvm",
         "-O3",
         "-DNDEBUG",
-        "-mprefer-vector-width=512",
         "-DEIGEN_VECTORIZE_GENERIC",
         "-flax-vector-conversions",  # Jax Mac wheel build server has int64_t != long
         "-fno-builtin",
         "-Wno-psabi",
         "-std=c++17",
-        "-fno-experimental-sanitize-metadata=all",
-        "-fno-sanitize=all",
     ] + copts
 
     # Disabled features to avoid instrumentations in the IR. ALL sanitizers must be disabled.
